src/extensions/default/bramble/lib/ConsoleInterfaceManager.js

```diff
@@ -0,0 +1,152 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Alexandru Ghiura
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ * /
+
+/*jslint vars: true, plusplus: true, nomen: true */
+/*global define, console, brackets, $, Mustache */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var AppInit = brackets.getModule("utils/AppInit"),
+        WorkspaceManager = brackets.getModule("view/WorkspaceManager"),
+        ExtensionUtils = brackets.getModule("utils/ExtensionUtils"),
+        PreferencesManager = brackets.getModule("preferences/PreferencesManager");
+
+    var panel
+        , icon = null
+        , panelHTML = require("text!htmlContent/console.html");
+
+    icon = $("<button class=\"editor-console-icon-indicator\" title=\"Click to toggle the JavaScript Console \">Console</button>");
```

Comment:
You will need to render this with `Mustache` and inject the string, something like https://github.com/mozilla/brackets/commit/5158dfb948088767219ba88d39b3295b14bb9264.  NOTE: we use the `brackets.getModule("Strings")` module vs. `require('strings')` for the extension.  Most extension authors are encourage to provide their own strings.  We do it differently, and put our strings into the main bracket's `Strings` module.  You'll eventually have to have the string go in Thimble's files, which is where it gets translated.  @gideonthomas can help you when you get there.